dataactcore/interfaces/function_bag.py

```diff
@@ -296,7 +296,7 @@ def mark_job_status(job_id, status_name):
 
     # if status is changed to finished for the first time, check dependencies
     # and add to the job queue as necessary
-    if old_status != 'finished' and status_name == 'finished':
+    if old_status != 'finished' and status_name == 'finished' and not skip_check:
```

Comment:
Cross-file will still get kicked off because the dependent job (the D file validation jobs) will be marked as finished which'll continue with the dependency check since `skip_check` by default is `False`. It'll still work normally (as it does now). 

The skip_check is primarily used for the D file upload jobs. If the API comes back with no data (empty D file), it'll skip the dependency checks for the upload job when marking it as `finished` so that it doesn't kick off the related validation job (since there's no data to validate).